pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -362,6 +367,115 @@ func (l *LibvirtDomainManager) asyncMigrate(vmi *v1.VirtualMachineInstance) {
 	}(l, vmi)
 }
 
+func getVMIEphemeralDisksTotalSize() *resource.Quantity {
+	var baseDir = "/var/run/kubevirt-ephemeral-disks/"
+	totalSize := int64(0)
+	err := filepath.Walk(baseDir, func(path string, f os.FileInfo, err error) error {
+		if !f.IsDir() {
+			totalSize += f.Size()
+		}
+		return err
+	})
+	if err != nil {
+		log.Log.Reason(err).Warning("failed to get VMI ephemeral disks size")
+		return &resource.Quantity{Format: resource.BinarySI}
+	}
+
+	return resource.NewScaledQuantity(totalSize, 0)
+}
+
+func getVMIMigrationDataSize(vmi *v1.VirtualMachineInstance) int64 {
+	var memory resource.Quantity
+
+	// Take memory from the requested memory
+	if v, ok := vmi.Spec.Domain.Resources.Requests[k8sv1.ResourceMemory]; ok {
+		memory = v
+	}
+	// In case that guest memory is explicitly set, override it
+	if vmi.Spec.Domain.Memory != nil && vmi.Spec.Domain.Memory.Guest != nil {
+		memory = *vmi.Spec.Domain.Memory.Guest
+	}
+
+	//get total data Size
+	if vmi.Status.MigrationMethod == v1.BlockMigration {
+		disksSize := getVMIEphemeralDisksTotalSize()
+		memory.Add(*disksSize)
+	}
+	return memory.ScaledValue(resource.Giga)
+}
+
+func liveMigrationMonitor(vmi *v1.VirtualMachineInstance, dom cli.VirDomain, l *LibvirtDomainManager) {
+	logger := log.Log.Object(vmi)
+	start := time.Now().UTC().Unix()
+	lastProgressUpdate := start
+	progressWatermark := int64(0)
+	// TODO:(vladikr) move to configMap
+	progressTimeout := int64(150)
+
+	// TODO:(vladikr) move to configMap
+	completionTimeoutPerGiB := int64(800)
+	acceptableCompletionTime := completionTimeoutPerGiB * getVMIMigrationDataSize(vmi)
```

Comment:
depending on the workload in the guest VM, this acceptableCompletionTime may or may not be very accurate the way it's calculated here. For the most part, the heuristic you came up with here seems like a good default.

Do you think it would make sense to let the user provide a total migration timeout value in the  migration api? The k8s Job api has something like this. They call it a "Deadline" I believe

Reply:
Yep, sure. I thought to move these parameters (including progresssTimeout) to kubevirt configMap. Do you think migration api is a better place for it?

Reply:
i'm not completely sure yet.  Do you think there's value in being able to set this on a per migration basis?

Reply:
Yeah. I think it's better to be able to configure this per migration. Updated :)